R/simulateHubbell.R

```diff
@@ -24,56 +24,56 @@
 #' \code{\link[miaSim:convertToSE]{convertToSE}}
 #'
 #' @examples
-#' x <- simulateHubbell(n_species = 8, M = 10, I = 1000, d = 50,
-#'                         m = 0.02, tend = 100)
+#' x <- simulateHubbell(n_species = 8, M = 10, carrying_capacity = 1000, k_events = 50,
+#'                         migration_p = 0.02, t_end = 100)
 #'
 #' @return \code{simulateHubbell} returns an abundance matrix with
 #' species abundance as rows and time points as columns
 #'
-#' @importFrom stats rbinom
-#' @importFrom stats rmultinom
+#' @importFrom stats rbinom rmultinom
 #' @importFrom MatrixGenerics colSums2
 #'
 #' @references Rosindell, James et al. "The unified neutral theory of
 #' biodiversity and biogeography at age ten." Trends in ecology & evolution
 #' vol. 26,7 (2011).
 #
 #' @export
-simulateHubbell <- function(n_species, M, I = 1000, d = 10, m = 0.02, tskip = 0,
-            tend, norm = FALSE){
+simulateHubbell <- function(n_species, M, carrying_capacity = 1000,
+                            k_events = 10, migration_p = 0.02, t_skip = 0,
+                            t_end, norm = FALSE){
 
             #input check
-            if(!all(vapply(list(n_species,M,I,d), .isPosInt,
+            if(!all(vapply(list(n_species,M,carrying_capacity,k_events), .isPosInt,
                     logical(1)))){
-                stop("n_species,M,I,d,tskip must be integer.")}
+                stop("n_species,M,carrying_capacity,k_events,t_skip must be integer.")}
 
             pbirth <- runif(n_species, min = 0, max = 1)
             pmigr <- runif(M, min = 0, max = 1)
             pbirth <- c(pbirth, rep(0, times = (M-n_species)))
             pbirth <- pbirth/sum(pbirth)
             pmigr <- pmigr/sum(pmigr)
-            com <- ceiling(I*pbirth)
-            if(sum(com)>I){
-                ind <- sample(seq_len(M), size = sum(com)-I, prob = 1-pbirth)
+            com <- ceiling(carrying_capacity*pbirth)
+            if(sum(com)>carrying_capacity){
+                ind <- sample(seq_len(M), size = sum(com)-carrying_capacity, prob = 1-pbirth)
                 com[ind] <- com[ind] -1
             }
 
-            tseries <- matrix(0, nrow = M, ncol = tend)
-            colnames(tseries) <- paste0("t", seq_len(tend))
+            tseries <- matrix(0, nrow = M, ncol = t_end)
+            colnames(tseries) <- paste0("t", seq_len(t_end))
             rownames(tseries) <- seq_len(M)
             com[which(com < 0)] <- 0
             tseries[,1] <- com
-            for (t in 2:tend){
+            for (t in 2:t_end){
```

Comment:
nowadays in R packages one should use `seq(2, t_end, 1)`, we may not get through the Bioc reviews using 2:n notation. Change here and everywhere. When it runs from 1:n then use `seq_len(n)`